### ISSUE TITLE
Using ngrok alternatives

### ISSUE THREAD

**Issue by dluc** [OfficeDev/TeamsFx#5045]
My team is developing several Teams bots and we use `teamsfx preview` to quickly test changes locally and avoid long build and deployment times. 

I noticed that the local preview uses `ngrok` to set up an SSL tunnel and expose localhost. This is a potential blocker because it exposes local machines through a third party service, and this third party has the ability to sniff the content moving through our bots. Even while developing, we are dealing with sensitive information that we don't want to travel through unverifiable systems.

We have our own private instance of *localtunnel* (https://github.com/localtunnel/localtunnel), a solution equivalent to ngrok, hosted in our cloud and using our own SSL certificates.

Is there a way to configure `teamsfx` to use our own SSL tunnel? 
If not, are you planning to add a similar feature to avoid trusting third parties?

**Comment by sffamily** [comment]
Hi @dluc your scenario is quite valid. As you pointed out the ngrok may encounter challenges if it is used in many enterprises with strict IT/network policies.

We will have ngrok alternative solution soon after our GA next week. It will be Microsoft 1P local tunneling solution. But as you pointed out, it may still encounter the similar challenge due to corp network policy. 

We'd like to learn about your custom solution further and we are confident to add support of custom local tunneling solutions. (Since we've been able to support ngrok and another local tunnel already.) 

I've started a Teams group chat with relevant folks on this topic in it to discuss with you.
